x-pack/legacy/plugins/ingest/server/rest_api/datasource/index.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import * as Joi from 'joi';
+import {
+  ReturnTypeCreate,
+  ReturnTypeUpdate,
+  ReturnTypeGet,
+  ReturnTypeList,
+} from '../../../common/types/std_return_format';
+import {
+  FrameworkRequest,
+  FrameworkRouteHandler,
+} from '../../libs/adapters/framework/adapter_types';
+import { ServerLibs, Datasource } from '../../libs/types';
+
+export const createGETDatasourceRoute = (libs: ServerLibs) => ({
+  method: 'GET',
+  path: '/api/ingest/datasource/{datasourceId}',
+  config: {},
+  handler: (async (
+    request: FrameworkRequest<{ params: { datasourceId: string } }>
+  ): Promise<ReturnTypeGet<any>> => {
+    const datasource = await libs.datasources.get(request.user, request.params.datasourceId);
+
+    return { item: datasource, success: true };
+  }) as FrameworkRouteHandler,
+});
+
+export const createGETDatasourcesRoute = (libs: ServerLibs) => ({
+  method: 'GET',
+  path: '/api/ingest/datasources',
```

Comment:
I wonder if we should follow here what ES does. For example for `_template` API if you use `GET _template/` you get all the templates. So the above would become `path: '/api/ingest/datasource'`.

Reply:
the current routes are consistent with our policy API patterns and various fleet APIs, so I will defer this for now